backend/main.py

```
@@ -14,42 +18,251 @@
+        if response.status_code != 200:
+            logger.error(f"HF API error: {response.status_code} - {response.text}")
+            response.raise_for_status()
+        
+        response_data = response.json()
+        logger.info("Successfully received response from HF API")
+        
+        return response_data
+        
+    except requests.exceptions.Timeout:
+        logger.error("Request to HF API timed out")
+        raise HTTPException(status_code=504, detail="Request timed out")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)